plugin/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/index.jelly

```diff
@@ -67,36 +67,13 @@ THE SOFTWARE.
                     <f:dropdownListBlock title="${%— Advanced/Deprecated —}"/>
                     <local:listSteps quasiDescriptors="${it.getQuasiDescriptors(true)}"/>
                 </f:dropdownList>
-                <j:set var="id" value="${h.generateId()}"/>
                 <f:block>
-                    <input type="button" value="${%Generate Pipeline Script}" onclick="handlePrototype_${id}(); return false" class="submit-button primary"/>
-                    <f:textarea id="prototypeText_${id}" readonly="true" style="margin-top: 10px" />
-                    <script>
-                        function handlePrototype_${id}() {
-                            buildFormTree(document.forms.config);
-                            // TODO JSON.stringify fails in some circumstances: https://gist.github.com/jglick/70ec4b15c1f628fdf2e9 due to Array.prototype.toJSON
-                            // TODO simplify when Prototype.js is removed
-                            var json = Object.toJSON ? Object.toJSON(JSON.parse(document.forms.config.elements.json.value).prototype) : JSON.stringify(JSON.parse(document.forms.config.elements.json.value).prototype);
-                            if (!json) {
-                                return; // just a separator
-                            }
-                            fetch('${rootURL}/${it.GENERATE_URL}', {
-                                method: 'post',
-                                headers: crumb.wrap({
-                                    "Content-Type": "application/x-www-form-urlencoded",
-                                }),
-                                body: new URLSearchParams({
-                                    json: json,
-                                }),
-                            }).then((rsp) => {
-                                if (rsp.ok) {
-                                    rsp.text().then((responseText) => {
-                                        document.getElementById('prototypeText_${id}').value = responseText;
-                                    });
-                                }
-                            });
-                        }
-                    </script>
+                    <input type="button" id="generatePipelineScript" value="${%Generate Pipeline Script}"
+                           class="submit-button primary"
+                           data-url="${rootURL}/${it.GENERATE_URL}"
+                           data-crumb="${h.getCrumb(request)}"/>
```

Comment:
I've added the jenkins crumb to avoid 403.
That's the easiest way, I found, to pass it to the javascript script.

Reply:
Could also just add a `CrumbExclusion` for https://github.com/jenkinsci/workflow-cps-plugin/blob/4e2b31c20833a35b08a00e9f5b8c3a042b9820df/plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java#L487 since it takes `POST` only to handle larger quantities of input but it is in fact read-only so not a CSRF threat. Anyway, this is the straightforward translation of the original.